packages/flutter_tools/lib/src/device.dart

```diff
@@ -948,6 +952,7 @@ class DebuggingOptions {
       if (route != null) '--route=$route',
       if (platformArgs['trace-startup'] as bool? ?? false) '--trace-startup',
       if (enableImpeller) '--enable-impeller',
+      if (serveObservatory) '--serve-observatory',
```

Comment:
I think this needs to be plumbed in more places, like here: lib/src/android/android_device.dart

and possibly for other devices as well?

Reply:
I'm not sure why we would need this plumbed through there since this doesn't actually need to be sent to the engine.

Reply:
If it doesn't need to be sent to the engine, then why is it in this list?

Reply:
Ah, I must have missed removing this entry. I was originally going to pipe this all through the engine and had a lot of the wiring done before changing to the private RPC approach.